### ISSUE TITLE
Error: Could not parse objectName on function grant

### ISSUE THREAD

**Issue by sfc-gh-pkommini** [Snowflake-Labs/terraform-provider-snowflake#1392]
**Provider Version**: 0.52.0

**Terraform Version**: 1.3.4

**Describe the bug**

My snowflake_function_grant looks like this:

```hcl
resource "snowflake_function_grant" "send_to_sentry_function_grant_usage" {
  provider = snowflake.my_role

  database_name = var.my_database
  schema_name   = var.my_schema
  function_name = snowflake_external_function.my_function.name

  # Function arguments
  arguments {
    name = "a"
    type = "VARCHAR(16777216)"
  }

  arguments {
    name = "b"
    type = "VARCHAR(16777216)"
  }

  arguments {
    name = "c"
    type = "VARCHAR(16777216)"
  }

  arguments {
    name = "d"
    type = "VARCHAR(16777216)"
  }

  arguments {
    name = "e"
    type = "VARCHAR(16777216)"
  }

  arguments {
    name = "f"
    type = "VARCHAR(16777216)"
  }

  return_type = "VARIANT"
  privilege   = "USAGE"
  roles       = var.my_function_usage_roles
}
```
on running `terraform apply` we get the following error:

```
╷
│ Error: Could not parse objectName: MY_FUNCTION(A VARCHAR(16777216), B VARCHAR(16777216), C VARCHAR(16777216), D VARCHAR(16777216), E VARCHAR(16777216), F VARCHAR(16777216)):VARIANT
│ 
│   with module.sentry_integration.snowflake_function_grant.send_to_sentry_function_grant_usage,
│   on ../../grants.tf line 1, in resource "snowflake_function_grant" "my_function_grant":
│    1: resource "snowflake_function_grant" "my_function_grant_usage" {
│ 
╵
```

**Expected behavior**
We expect the code to create the grant without error.

**Code samples and commands**
Please add code examples and commands that were run to cause the problem.

**Comment by sfc-gh-hachouraria** [comment]
Looking at [the internal regex](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/792665f7fea6cbe3c5df4906ba298efd2f6727a1/pkg/resources/grant_helpers.go#L441-L446) that’s failing, the issue appears to be the use of `(…)` characters in the data types of arguments. If you simplify the type names it appears to go through:

```diff
  arguments {
    name = "a"
-    type = "VARCHAR(16777216)"
+    type = "VARCHAR"
  }
```

**Comment by sfc-gh-swinkler** [comment]
fix released as part of 0.56.1. Note that some changes were made to the grant resource to make it a little easier to use. Example

```
  resource "snowflake_function_grant" "test_proc_grant" {
  database_name = snowflake_database.my_database.name
  schema_name   = snowflake_schema.my_schema.name
    function_name = snowflake_function.test_func.name
    argument_data_types = [
      "VARCHAR(16777216)"
    ]
    privilege = "USAGE"
    roles     = [snowflake_role.my_role.name]
  }
```
